sdk/dynatrace/Azure.ResourceManager.DynaTrace/tests/Scenario/MonitorTestBase.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.DynaTrace.Models;
+using Azure.ResourceManager.Resources;
+
+namespace Azure.ResourceManager.DynaTrace.Tests
+{
+    public class MonitorTestBase : DynaTraceManagementTestBase
+    {
+        protected ResourceGroupResource _resourceGroup;
+        protected GenericResourceCollection _genericResourceCollection;
+
+        public MonitorTestBase(bool isAsync) : base(isAsync)
+        {
+        }
+
+        public MonitorTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync, mode)
+        {
+        }
+
+        protected async Task<MonitorResourceCollection> GetMonitorResourceCollectionAsync()
+        {
+            _genericResourceCollection = Client.GetGenericResources();
+            _resourceGroup = await CreateResourceGroupAsync();
+            return _resourceGroup.GetMonitorResources();
+        }
+
+        protected async Task<MonitorResource> CreateMonitorResourceAsync(string monitorName)
+        {
+            var collection = await GetMonitorResourceCollectionAsync();
+            var input = GetMonitorInput();
+
+            var lro = await collection.CreateOrUpdateAsync(WaitUntil.Completed, monitorName, input);
+            return lro.Value;
+        }
+
+        public MonitorResourceData GetMonitorInput()
+        {
+            var aadDomains = new List<string>();
+            aadDomains.Add("SDKTest");
+
+            return new MonitorResourceData(DefaultLocation)
+            {
+                UserInfo = new UserInfo
+                {
+                    FirstName = "Divyansh",
+                    LastName = "Agarwal",
+                    PhoneNumber = "1234567890",
+                    Country = "US",
+                    EmailAddress = "agarwald@microsoft.com"
+                },
+                PlanData = new PlanData
+                {
+                    UsageType = "COMMITTED",
+                    BillingCycle = "Monthly",
+                    PlanDetails = "azureportalintegration_privatepreview",
+                    EffectiveOn = System.DateTimeOffset.Now
```

Comment:
Please do not use this expression, in CI pipeline we use the record json files to verify your test. If you use `Now`, this property will be different from the value in your record.

Reply:
Updated this. Thanks for this suggestion, because of this our test were failing in the Playback mode